### PR TITLE
[java native][dotnetcore] Implement QueryParameter deepObject style

### PR DESCRIPTION
**Pull request by Reinhard-PTV**
[java native][dotnetcore] Implement QueryParameter deepObject style for client generators java native and dotnetcore.

https://github.com/OpenAPITools/openapi-generator/issues/469

@mandrean (2017/08) @frankyjuang (2019/09) @shibayan (2020/02)
@bbdouglas (2017/07) @sreeshas (2017/08) @jfiala (2017/08) @lukoyanov (2017/09) @cbornet (2017/09) @jeff9finger (2018/01) @karismann (2019/03) @Zomzog (2019/04) @lwlee2608 (2019/10) @nmuesch (2021/01)

Please have a look at the new tests:

- modules/openapi-generator/src/test/java/org/openapitools/codegen/java/JavaClientDeepObjectTest.java
- modules/openapi-generator/src/test/java/org/openapitools/codegen/csharpnetcore/CSharpNetCoreClientDeepObjectTest.java

with the test document modules/openapi-generator/src/test/resources/3_0/deepobject.yaml

The urls for the deepObject query parameters are generated as specified in 

https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.3.md#style-values

### PR checklist
 
- [ x ] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [ x ] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [ x ] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [ x ] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `5.1.x`, `6.0.x`
- [ x ] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

**Comment by Reinhard-PTV** [comment]
The Shippable build fails with:

> W: GPG error: http://binaries.erlang-solutions.com/debian xenial Release: The following signatures were invalid: BADSIG D208507CA14F4FCA Erlang Solutions Ltd. <packages@erlang-solutions.com>
W: The repository 'http://binaries.erlang-solutions.com/debian xenial Release' is not signed.
W: An error occurred during the signature verification. The repository is not updated and the previous index files will be used. GPG error: https://downloads.apache.org/cassandra/debian 311x InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY E91335D77E3E87CB
W: Failed to fetch http://www.apache.org/dist/cassandra/debian/dists/311x/InRelease  The following signatures couldn't be verified because the public key is not available: NO_PUBKEY E91335D77E3E87CB
E: Failed to fetch http://binaries.erlang-solutions.com/debian/dists/xenial/contrib/binary-amd64/Packages.bz2  Hash Sum mismatch

Can you please assist?

**Comment by wing328** [comment]
cc @OpenAPITools/generator-core-team

**Review by wing328 — APPROVED** [review]
Tested locally an didn't spot any issue

File: modules/openapi-generator/src/test/java/org/openapitools/codegen/csharpnetcore/CSharpNetCoreClientDeepObjectTest.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2018 OpenAPI-Generator Contributors (https://openapi-generator.tech)
+ * Copyright 2018 SmartBear Software
```
Comment (wing328) [inline modules/openapi-generator/src/test/java/org/openapitools/codegen/csharpnetcore/CSharpNetCoreClientDeepObjectTest.java:3]: Will remove this line in one of my upcoming PRs

Ref: https://opensource.stackexchange.com/questions/7300/copyright-notice-in-the-file-header-apache-v2-license

File: modules/openapi-generator/src/test/java/org/openapitools/codegen/java/JavaClientDeepObjectTest.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2018 OpenAPI-Generator Contributors (https://openapi-generator.tech)
+ * Copyright 2018 SmartBear Software
```
Comment (wing328) [inline modules/openapi-generator/src/test/java/org/openapitools/codegen/java/JavaClientDeepObjectTest.java:3]: Will remove this line in one of my upcoming PRs

Ref: https://opensource.stackexchange.com/questions/7300/copyright-notice-in-the-file-header-apache-v2-license

**Pull request merged**